targets/frontend/src/components/contributions/answers/Answer.mutation.ts

```diff
@@ -3,35 +3,64 @@ import { OperationResult, useMutation } from "urql";
 import { Answer } from "../type";
 
 export const contributionAnswerUpdateMutation = `
-mutation contributionAnswerUpdate($id: uuid!, $content:String, $otherAnswer: String, $status: statustype!, $userId: uuid!) {
-    update_contribution_answers_by_pk(
-      pk_columns: {
-        id: $id
-      }
-      _set: {
-        content: $content
-        other_answer: $otherAnswer
-      }
-    ) {
-        __typename
-    }
-    insert_contribution_answer_statuses_one(object:{status:$status, user_id:$userId,answer_id:$id}) {
-      id, created_at
-    }
+mutation contributionAnswerUpdate($id: uuid!, $content: String, $otherAnswer: String, $status: statustype!, $userId: uuid!, $kali_references: [contribution_answer_kali_references_insert_input!]!, $legi_references: [contribution_answer_legi_references_insert_input!]!, $other_references: [contribution_answer_other_references_insert_input!]!, $cdtn_documents: [contribution_answer_cdtn_documents_insert_input!]!) {
+  update_contribution_answers_by_pk(pk_columns: {id: $id}, _set: {content: $content, other_answer: $otherAnswer}) {
+    __typename
+  }
+  insert_contribution_answer_statuses_one(object: {status: $status, user_id: $userId, answer_id: $id}) {
+    id
+    created_at
+  }
+  delete_contribution_answer_kali_references(where: {answer_id: {_eq: $id}}) {
+    affected_rows
+  }
+  insert_contribution_answer_kali_references(objects: $kali_references, on_conflict: {constraint: answer_kali_references_pkey}) {
+    affected_rows
+  }
+  delete_contribution_answer_legi_references(where: {answer_id: {_eq: $id}}) {
+    affected_rows
+  }
+  insert_contribution_answer_legi_references(objects: $legi_references, on_conflict: {constraint: answer_legi_references_pkey}) {
+    affected_rows
+  }
+  delete_contribution_answer_cdtn_documents(where: {answer_id: {_eq: $id}}) {
+    affected_rows
+  }
+  insert_contribution_answer_cdtn_documents(objects: $cdtn_documents, on_conflict: {constraint: answer_cdtn_documents_pkey}) {
+    affected_rows
   }
+  delete_contribution_answer_other_references(where: {answer_id: {_eq: $id}}) {
+    affected_rows
+  }
+  insert_contribution_answer_other_references(objects: $other_references, on_conflict: {constraint: answer_other_references_pkey}) {
+    affected_rows
+  }
```

Comment:
dommage qu'il est pas d'upsert sur les références:'(

c'est vrai que la logique métier de transaction se trouve là avec graphql, mais bon, on a pas trop d'autres choix, hormis peut-être faire de la concatenation de string avec des variable qui permet de mieux visualiser ces calls, mais bon, c'est assez inhérent à hasura d'avoir de la logique comme ça

Reply:
Après la logique de supprimer et recréer n'est pas déconnante ici avec ce lien one-to-many où le many n'a pas de lien externe.

Reply:
Après niveau lisibilité, je suis également plus habitué à voir ça dans du code :)